pkg/kp/rollstore/consul_store.go

```diff
@@ -171,6 +171,20 @@ func (s consulStore) checkForConflictingUpdates(rcIDs rc_fields.IDs) error {
 	return nil
 }
 
+// Best-effort applies labels to the specified RC. It's the callers
+// responsibility to put things back the way they were in an error scenario due
+// to the variety of situations this is called in.
+func (s consulStore) labelNewRC(rcID rc_fields.ID, rcLabels klabels.Set) error {
+	for k, v := range rcLabels {
+		err := s.labeler.SetLabel(labels.RC, rcID.String(), k, v)
```

Comment:
although it'll be weird because you don't want to overwrite any other labels, it'd more be like batch apply and batch delete